File: 344-ReverseString/344-ReverseString.py
# Last updated: 6/25/2026, 8:37:46 AM
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        strLength = len(s)
        startPtr = 0
        endPtr = strLength - 1
        while(startPtr < endPtr):
            t = s[startPtr]
            s[startPtr] = s[endPtr]
            s[endPtr] = t
            startPtr += 1
            endPtr -= 1

File: 344-ReverseString/344-ReverseString.java
// Last updated: 6/25/2026, 8:37:41 AM
class Solution {
    public void reverseString(char[] s) {
        int head = 0;
        int tail = s.length - 1;
        while(head < tail){
            char temp = s[head];
            s[head] = s[tail];
            s[tail] = temp;
            head += 1;
            tail -= 1;
        }
    }
}